app/services/buda_user_service.rb

```diff
@@ -6,19 +6,19 @@ def generate_invoice(bitcoins_amount)
     url ='https://www.buda.com/api/v2/lightning_network_invoices'
     request_type = 'POST'
     satoshis = satoshi_price(bitcoins_amount)
-    body ={ amount_satoshis: satoshis, currency: 'BTC' }.to_json
+    body = { amount_satoshis: satoshis, currency: 'BTC' }.to_json
     nonce = generate_nonce
-    headers = headers(@api_key, @api_secret, nonce, request_type, path ,body)
+    headers = headers(@api_key, @api_secret, nonce, request_type, path, body)
     post_request(url, body, headers)
   end
 
   def pay_invoice(bitcoins_amount, invoice_code, simulate)
-    path = '/api/v2/currencies/BTC/withdrawals'
-    url ='https://www.buda.com/api/v2/currencies/BTC/withdrawals'
+    path = '/api/v2/reserves/ln-btc/withdrawals'
+    url ='https://www.buda.com/api/v2/reserves/ln-btc/withdrawals'
```

Comment:
Separaría esto para no repetir el path en los dos lados y lo mismo en generate_invoice
```suggestion
    base='https://www.buda.com'
    path = '/api/v2/reserves/ln-btc/withdrawals'
    url = base + path
```